scripts/install_backend_python_libs.py

```diff
@@ -330,25 +356,24 @@ def validate_metadata_directories():
     # directory name. Otherwise, we would need to check every permutation of
     # the casing for metadata directories generated with the naming convention:
     # <library_name>-<library-version>.
-    file_or_directory_names = set(
+    directory_names = set(
```

Comment:
normalized_directory_names

Also, I wonder about something. Could you get rid of one layer of normalization by forcing all the dirnames in the third-party python libs dir to be lowercase at installation time, or is that not possible?

Reply:
Done. I don't have control over the naming of the directories, I checked and pip doesn't support an option to rename the metadata files.

Reply:
Do the (original) library names always match exactly the names of the metadata files (is that guaranteed)?

Reply:
They match the library name but there is no guarantee that the casing matches. That's why I just normalize everything. The documentation only mentions that the directory names are in the format <library-name>-<library-version> but there are multiple cases -> in our current repo all of the names in the requirements.txt file are lowercase but when we install it, there are some that matches the original casing of the default library name and others that don't.

Reply:
Then add that (factual) info here, in the comment, with specific examples, so that it's clear to future developers why we need all this. I.e. provide a reference to explain why they match the library name (so that it's clear why lowercasing is a sufficient guarantee of correctness), and specific examples to show why casing doesn't work (so that it's clear why we can't have a more specific guarantee).

Reply:
Done.